datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxAttachDetachUtils.java

```diff
@@ -21,13 +21,14 @@ public void updateClusterNameAndCrn(SdxCluster sdxCluster, String newName, Strin
         sdxCluster.setClusterName(newName);
         sdxCluster.setOriginalCrn(sdxCluster.getCrn());
         sdxCluster.setCrn(newCrn);
+        sdxCluster.setStackCrn(newCrn);
     }
 
-    public void updateStack(SdxCluster cluster, String originalName) {
+    public void updateStack(String originalName, String newName, String newCrn) {
         String initiatorUserCrn = ThreadBasedUserCrnProvider.getUserCrn();
         ThreadBasedUserCrnProvider.doAsInternalActor(() ->
                 stackV4Endpoint.updateNameAndCrn(
-                        0L, originalName, initiatorUserCrn, cluster.getClusterName(), cluster.getCrn()
+                        0L, originalName, initiatorUserCrn, newName, newCrn
```

Comment:
What's wrong with the original implementation? Why is this refactor done?

Reply:
I modified this in order to add https://github.com/hortonworks/cloudbreak/pull/12228/files/d241d343c00b8a2d544d86ba3f51490d90ca7a4f#diff-4480719b7166ebd935c5cf33cb173cf2a4b6899df15f901679b078583cdb7af5R112

Specifically, I wanted to make sure that if recovery fails, everything is brought back to the point before recovery started (if possible).